### ISSUE TITLE
Idempotence plugin was not found or did not provide the required information.

### ISSUE THREAD

**Issue by pgporada** [metacloud/molecule#291]
Molecule version
```
1.8.3
```

Host details
```
Ubuntu 16.04
Linux laptappy 4.4.0-28-generic #47-Ubuntu SMP Fri Jun 24 10:09:13 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

molecule.yml
```
---
ansible:
  playbook: tests/test.yml

docker:
  containers:
  - name: ansible-jinja-examples-01
    ansible_groups:
      - group1
    image: centos
    image_version: latest
```

I can run `molecule converge` but not `molecule idempotence`. When I try to, I receive the following message.

```
DEBUG: ANSIBLE ENVIRONMENT
ANSIBLE_CALLBACK_PLUGINS: /usr/share/molecule/ansible/plugins/callback/idempotence
ANSIBLE_CONFIG: .molecule/ansible.cfg
ANSIBLE_FORCE_COLOR: 'false'
ANSIBLE_HOST_KEY_CHECKING: 'false'
ANSIBLE_NOCOLOR: 'true'
ANSIBLE_SSH_ARGS: -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ControlMaster=auto
  -o ControlPersist=60s

DEBUG: ANSIBLE PLAYBOOK
/usr/local/bin/ansible-playbook tests/test.yml --inventory-file=.molecule/ansible_inventory --sudo --connection=docker --limit=all --user=root --timeout=30 --diff
--> Starting Ansible Run ...
Idempotence test failed.
The idempotence plugin was not found or did not provide the required information. Therefore the failure details cannot be displayed.
```

**Comment by retr0h** [comment]
Does the file exist?

`/usr/share/molecule/ansible/plugins/callback/idempotence`

**Comment by pgporada** [comment]
@retr0h 
It does not. I installed molecule per the README.md.

```
$ ll /usr/share/molecule/ansible/plugins/callback/idempotence
ls: cannot access '/usr/share/molecule/ansible/plugins/callback/idempotence': No such file or directory

$ sudo ls -al /usr/share/molecule/ansible/plugins/callback/idempotence
ls: cannot access '/usr/share/molecule/ansible/plugins/callback/idempotence': No such file or directory
```

Additionally,  bash completion does not work and the completion file does not exist.

**Comment by kireledan** [comment]
I've had this occur - it has to do with the ansible installation on specifc distros. I wish I knew how to resolve this because it's supposed to be bundled with ansible.

**Comment by retr0h** [comment]
Lets see what @rgreinho has to say.  He implemented both of those features.

**Comment by rgreinho** [comment]
Yes, if you are using OS X, it looks like the files are not extract at the right location. 

For the bach completion, here is the documentation to install it: https://github.com/metacloud/molecule/blob/master/docs/source/usage.rst#os-x-users

You can proceed in a similar way to add the idempotence plugin.

**Comment by pporada-gl** [comment]
@rgreinho 
I am using the following host system
```
Ubuntu 16.04
Linux laptappy 4.4.0-28-generic #47-Ubuntu SMP Fri Jun 24 10:09:13 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by pgporada** [comment]
@rgreinho 
I am using the following host system
```
Ubuntu 16.04
Linux laptappy 4.4.0-28-generic #47-Ubuntu SMP Fri Jun 24 10:09:13 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by rgreinho** [comment]
Then I guess you installed molecule in a venv. In that case, the venv will be the prefix of installation paths. 

In my case, my venv is ``~/.virtualenvs/molecule-extras``, therefore I can find the plugin there: ``~/.virtualenvs/molecule-extras/share/molecule/ansible/plugins/callback/idempotence/idempotence.py``

Same thing for the bash completion: ``/Users/regreinh/.virtualenvs/molecule-extras/etc/bash_completion.d/molecule.bash-completion.sh``. You must enable it by hand in this case as well.

**Comment by pgporada** [comment]
@rgreinho 
I have no virtual env.
```
$ which python
/usr/bin/python
$ ll /usr/bin/python
lrwxrwxrwx 1 root root 9 Dec  9  2015 /usr/bin/python -> python2.7*
```

Even when installing it as follows, the bash completion and idempotence plugin is missing.
```
$ pip install --user molecule
# or
$ sudo pip install molecule
```

**Comment by rgreinho** [comment]
What about using the ``find``command to find where the files are? They are part of the package, therefore they must be somewhere on your system.

**Comment by pgporada** [comment]
@rgreinho 
No luck friend. I've searched the entire filesystem.
```
$ find / ! -readable -prune -type f -iname idempotence.py 
find: ‘/proc/11087/task/11087/fd/5’: No such file or directory
find: ‘/proc/11087/task/11087/fdinfo/5’: No such file or directory
find: ‘/proc/11087/fd/6’: No such file or directory
find: ‘/proc/11087/fdinfo/6’: No such file or directory

$ find / ! -readable -prune -type f -iname molecule.bash-completion.sh
find: ‘/proc/11160/task/11160/fd/5’: No such file or directory
find: ‘/proc/11160/task/11160/fdinfo/5’: No such file or directory
find: ‘/proc/11160/fd/6’: No such file or directory
find: ‘/proc/11160/fdinfo/6’: No such file or directory
```

**Comment by pgporada** [comment]
Bah, find doesn't traverse filesystem boundaries. Sorry about that. Regardless, I can't find either of those files on my system no matter how I `find`

**Comment by rgreinho** [comment]
@pgporada I'm really sorry, I am not able to reproduce the issue. Whichever way I install molecule, I can find the plugin and completion files on my system. 

Maybe another member can try to reproduce this issue?

**Comment by retr0h** [comment]
@pgporada what version of python?

**Comment by retr0h** [comment]
@pgporada I'll spin up a Xenial vm and take a look later.

**Comment by retr0h** [comment]
Appears to have worked for me.

```
root@mcp1:~# python --version
Python 2.7.12
root@mcp1:~# molecule --version
WARNING:vagrant:The Vagrant executable cannot be found. Please check if it is in the system path.
1.8.3
root@mcp1:~# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04 LTS
Release:	16.04
Codename:	xenial
```

```
root@mcp1:/tmp# molecule init foo --docker
WARNING:vagrant:The Vagrant executable cannot be found. Please check if it is in the system path.
--> Initializing role foo...
cSuccessfully initialized new role in ./foo
droot@mcp1:/tmp# cd foo
root@mcp1:/tmp/foo# molecule test
--> Destroying instances ...
--> Checking playbooks syntax ...

playbook: playbook.yml
--> Creating instances ...
--> Creating Ansible compatible image of ubuntu:latest ...
Building ansible compatible image ...
	Step 1 : FROM ubuntu:latest
	Pulling fs layer ...
.....
Finished building molecule_local/ubuntu:latest
Creating container foo-01 with base image ubuntu:latest ...
Container created.
Creating container foo-02 with base image ubuntu:latest ...
Container created.
--> Starting Ansible Run ...

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [foo-01]
ok: [foo-02]

PLAY RECAP *********************************************************************
foo-01                     : ok=1    changed=0    unreachable=0    failed=0
foo-02                     : ok=1    changed=0    unreachable=0    failed=0

--> Idempotence test in progress (can take a few minutes)...
--> Starting Ansible Run ...
Idempotence test passed.
--> Executing testinfra tests found in tests/.
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /tmp/foo/tests, inifile:
plugins: testinfra-1.4.1, xdist-1.14
collected 2 itemss

tests/test_default.py ..

=========================== 2 passed in 0.78 seconds ===========================
No serverspec tests found in spec/.
--> Destroying instances ...
Stopping container foo-01 ...
Removed container foo-01.
Stopping container foo-02 ...
Removed container foo-02.
```

**Comment by rgreinho** [comment]
Closing as we cannot reproduce the issue.

**Comment by retr0h** [comment]
I believe this is still an issue, see #437.

**Comment by retr0h** [comment]
Released molecule [1.10.3](https://pypi.python.org/pypi/molecule) to pypi to address this bug. 

There is no change in workflow.  Molecule still reports if a converge was idempotent or not.  However, it no longer reports which task(s) are not idempotent.  This greatly simplifies the code and issues around this.
